### PR TITLE
Twofish

### PR DESCRIPTION
**Pull request by SamProtas**
Implements 128bit key Twofish BlockCipher as requested here:
https://github.com/haskell-crypto/cryptonite/issues/33

I used the Golang Twofish implementation as a reference and included KAT's from the inventor's website.

Let me know if there's anything more I can do to get this merged.

### CODE REVIEW

File: Crypto/Cipher/Twofish/Primitive.hs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+blockSize = 16
+
+mdsPolynomial, rsPolynomial :: Word32
+mdsPolynomial = 0x169 -- x^8 + x^6 + x^5 + x^3 + 1, see [TWOFISH] 4.2
+rsPolynomial = 0x14d  -- x^8 + x^6 + x^3 + x^2 + 1, see [TWOFISH] 4.3
+
+data Twofish = Twofish { s :: ([Word32], [Word32], [Word32], [Word32])
+                       , k :: [Word32] }
+
+-- CONFIRMED
```
Comment (vincenthz) [inline Crypto/Cipher/Twofish/Primitive.hs]: ?

Comment (SamProtas) [reply Crypto/Cipher/Twofish/Primitive.hs]: Just notes to myself during development. Accidentally committed.

**Comment by SamProtas** [comment]
FYI I've started looking at adding support for 192 and 256 bit keys. I don't think it'll be too time consuming so I may have something to add to this PR tomorrow if I find time to work on it.

**Comment by vincenthz** [comment]
thanks looking good

**Pull request merged**